flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/compiler/NFACompiler.java

```diff
@@ -150,6 +160,59 @@ long getWindowTime() {
 		}
 
 		/**
+		 * Check pattern after match skip strategy.
+		 */
+		private void checkPatternSkipStrategy() {
+			AfterMatchSkipStrategy afterMatchSkipStrategy = currentPattern.getAfterMatchSkipStrategy();
+			if (afterMatchSkipStrategy.getStrategy() == AfterMatchSkipStrategy.SkipStrategy.SKIP_TO_FIRST ||
+				afterMatchSkipStrategy.getStrategy() == AfterMatchSkipStrategy.SkipStrategy.SKIP_TO_LAST) {
+				Pattern<T, ?> pattern = currentPattern;
+				while (!pattern.getName().equals(afterMatchSkipStrategy.getPatternName())) {
+					pattern = pattern.getPrevious();
+				}
+				// pattern name match check.
+				if (pattern == null) {
```

Comment:
It is always true. If `pattern.getPrevious` is `null`. The `while` loop will throw `NullPointerException`.

Missing test for this case.